test/test_unix_echo_server.rs

```diff
@@ -163,21 +154,21 @@ impl EchoClient {
                 debug!("CLIENT : We read {} bytes!", r);
 
                 // prepare for reading
-                let mut buf = buf.flip();
 
-                debug!("CLIENT : buf = {:?} -- rx = {:?}", buf.bytes(), self.rx.bytes());
-                while buf.has_remaining() {
-                    let actual = buf.read_byte().unwrap();
-                    let expect = self.rx.read_byte().unwrap();
+                debug!("CLIENT : buf = {:?} -- rx = {:?}", buf.bytes(), self.rx);
+                for actual in buf.iter() {
+                    let expect = self.rx[0];
+                    self.rx = &self.rx[1..];
 
-                    assert!(actual == expect, "actual={}; expect={}", actual, expect);
+                    assert!(*actual == expect, "actual={}; expect={}", actual, expect);
                 }
 
-                self.mut_buf = Some(buf.flip());
+                buf.clear();
```

Comment:
If I'm reading this correctly, this `clear` is fine because the `for` loop above checks every element.

If I am incorrect, please let me know :)

Reply:
Yeah that's the highest layer, where we received a "full message" and check if it's what we expected.